apps/site/lib/site_web/controllers/schedule/line/helpers.ex

```diff
@@ -80,10 +80,36 @@ defmodule SiteWeb.ScheduleController.Line.Helpers do
   defp do_get_branch_route_stops(route, direction_id, route_pattern_id) do
     route.id
     |> get_route_patterns(direction_id, route_pattern_id)
-    |> Enum.filter(&by_typicality(&1, route_pattern_id))
+    |> filter_by_min_typicality()
+    # The below row is for testing purposes for the ferry ONLY
+    # This is needed until a specific routepattern has a distinct typicality
+    |> temporary_hh_ferry_only_func(route.id)
     |> Enum.map(&stops_for_route_pattern/1)
   end
 
+  @spec filter_by_min_typicality([RoutePattern.t()]) :: [RoutePattern.t()]
+  defp filter_by_min_typicality(route_patterns) do
+    route_patterns
+    |> Enum.reduce({nil, []}, fn
+      %RoutePattern{typicality: typicality} = r, {t, a} when typicality < t -> {typicality, [r]}
+      %RoutePattern{typicality: typicality} = r, {typicality, a} -> {typicality, [r | a]}
+      _, acc -> acc
+    end)
```

Comment:
I'll admit I'm _still_ trying to wrap my head around how this works, but I think I understand what it's supposed to do? Kind of like a mix of Enum.group_by typicality and sorting on whatever the lowest typicality is?

Reply:
Whoops sorry for the duplicate comment, I had thought Github disappeared this one

Reply:
Yeah, you're right - iterating over the list, if a typicality is found that's less than the "t" typicality currently in the accumulator, then the list of "lowest typicality route patterns" currently in the accumulator will be ditched and replaced by this new route pattern that was just found.  I should expand those variables beyond just r / t / a, but the lines were getting verbose